src/instruction.rs

```diff
@@ -0,0 +1,268 @@
+//! Program instruction types.
+
+use solana_program::{
+    instruction::{AccountMeta, Instruction},
+    program_error::ProgramError,
+    pubkey::Pubkey,
+};
+
+/// Instructions supported by the Paladin Rewards program.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum PaladinRewardsInstruction {
+    /// Configures rewards for a mint that has been configured with the rewards
+    /// program as a transfer hook program.
+    ///
+    /// This instruction will:
+    ///
+    /// - Initialize a rewards account for the mint (distribution account),
+    ///   configured with the distribution addresses:
+    ///   - Active rewards
+    ///   - Piggy bank
+    ///   - Staked PAL rewards
+    /// - Initialize an active rewards account for the mint.
+    /// - Initialize the required accounts for the transfer hook.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    /// 0. `[w]` Distribution account.
+    /// 1. `[w]` Active rewards account.
+    /// 2. `[w]` Transfer hook extra account metas account.
+    /// 3. `[ ]` Token mint.
+    /// 4. `[s]` Mint authority.
+    InitializeMintRewardInfo {
+        piggy_bank_address: Pubkey,
+        staked_rewards_address: Pubkey,
```

Comment:
staked rewards need to be separate here so that they can be sent to the governance / some third-party program?

Reply:
ah this is configuring the third-party address to send staked rewards to

Reply:
Yep. I figured unless we need to do some checks on the account, we can just pass the address directly, to store in the rewards PDA.